render_delegate/basis_curves.cpp

```diff
@@ -198,12 +200,14 @@ void HdArnoldBasisCurves::Sync(
                     } else if (desc.value.IsHolding<VtVec3fArray>()) {
                         const auto& v = desc.value.UncheckedGet<VtVec3fArray>();
                         auto* arr = AiArrayAllocate(v.size(), 1, AI_TYPE_VECTOR2);
-                        std::transform(
-                            v.begin(), v.end(), static_cast<GfVec2f*>(AiArrayMap(arr)),
-                            [](const GfVec3f& in) -> GfVec2f {
-                                return {in[0], in[1]};
-                            });
-                        AiArrayUnmap(arr);
+                        if (!v.empty()) {
```

Comment:
question ... would `if (AiArrayGetDataSize(arr)) {` also work ? in case the allocation failed ...

Reply:
maybe, but since we explicitely set 1 key, the issue should only happen if v.size is 0, so it should be equivalent